include/alpaka/core/UniqueId.hpp

```diff
@@ -73,13 +73,13 @@ namespace alpaka
                 static constexpr std::size_t value = N;
             };
 
-#ifdef BOOST_COMP_MSVC
+#if (BOOST_COMP_MSVC || __CUDACC__)
```

Comment:
I tested `nvcc` again and the second pass not compile, therefore the `MSVC` workeround is used. The problem is that `noexcept(f())` always returned `false` (NVIDIA implements it like descriped [here](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2010/n3050.html#operator-noexcept-expression)) It defines that it must evaluted always to `false`